### ISSUE TITLE
[BUG] Volume with `robustness is unknown` is not reflect under Volume page

### ISSUE THREAD

**Issue by kaxing** [longhorn/longhorn#3291]
## Describe the bug

When attached note got deleted, the Unkown volume state is not shown.

## To Reproduce

1. Create a volume 1
2. Attach volume to node A
3. Update the replica count of volume 1 to 1
4. Delete all replicas and keep only one replica that is not on the same node with the volume 1
5. Delete the node A from kubectl or kubernetes dashboard
6. Goto the longhorn volume UI
7. Click into volume 1
8. Following error message will shown
```
fail to list snapshot: cannot get client for volume pvc-61195983-4b27-42a7-8b1e-e8238552b63e: engine is not running
```

## Expected behavior

Detail of _volume 1_ is reflecting the Unkown state instead of attached and giving error of other situation(supposedly, the volume state should > snapshot status)

Manually reattach _volume 1_ to other working node can resolve the volume healthiness to normal.

## Environment

 - Longhorn version: v1.1.3-rc1, master-head, v1.1.2
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): kubectl
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: rke1 v1.18
   - Number of management node in the cluster: 1
   - Number of worker node in the cluster: 3 + 1

## Additional context
`$ kubectl describe volume pvc-5f0cd541-d28c-40f8-a42e-0a2cb4aec7a6 -n longhorn-system` is showing relevant event info:
```
Events:
  Type     Reason              Age                From                               Message
  ----     ------              ----               ----                               -------
  Normal   Detached            24m                longhorn-volume-controller         volume pvc-5f0cd541-d28c-40f8-a42e-0a2cb4aec7a6 has been detached
  Normal   Start               24m                longhorn-kubernetes-pv-controller  Persistent Volume pvc-5f0cd541-d28c-40f8-a42e-0a2cb4aec7a6 started to use/reuse Longhorn volume pvc-5f0cd541-d28c-40f8-a42e-0a2cb4aec7a6
  Warning  DetachedUnexpectly  21m                longhorn-volume-controller         Engine of volume pvc-5f0cd541-d28c-40f8-a42e-0a2cb4aec7a6 dead unexpectedly, reattach the volume
  Normal   Remount             21m                longhorn-volume-controller         Volume pvc-5f0cd541-d28c-40f8-a42e-0a2cb4aec7a6 requested remount at 2021-11-22T10:30:13Z
  Normal   Detached            21m                longhorn-volume-controller         volume pvc-5f0cd541-d28c-40f8-a42e-0a2cb4aec7a6 has been detached
  Normal   Attached            21m (x2 over 24m)  longhorn-volume-controller         volume pvc-5f0cd541-d28c-40f8-a42e-0a2cb4aec7a6 has been attached to ip-A-B-C-D
  Warning  Unknown             2m49s              longhorn-volume-controller         volume pvc-5f0cd541-d28c-40f8-a42e-0a2cb4aec7a6 robustness is unknown
```
Related issue: #1451

**Comment by khushboo-rancher** [comment]
@kaxing After waiting for sometime, did the error go?

**Comment by kaxing** [comment]
@khushboo-rancher the "Attached" state remain quite a bit time(around 30mins) from the UI, hence the error can be reproduced easily by click into the volume detail page.

I will double check the http/wss responses.

**Comment by kaxing** [comment]
The response from _/v1/volumes_ still shows attached,
<Details>

<Summary>(expand to see the full json of the unkown volume) </Summary>

```json
        {
            "accessMode": "rwo",
            "actions":
            {
                "activate": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=activate",
                "attach": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=attach",
                "cancelExpansion": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=cancelExpansion",
                "detach": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=detach",
                "engineUpgrade": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=engineUpgrade",
                "pvCreate": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=pvCreate",
                "pvcCreate": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=pvcCreate",
                "recurringUpdate": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=recurringUpdate",
                "replicaRemove": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=replicaRemove",
                "snapshotBackup": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=snapshotBackup",
                "snapshotCreate": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=snapshotCreate",
                "snapshotDelete": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=snapshotDelete",
                "snapshotGet": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=snapshotGet",
                "snapshotList": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=snapshotList",
                "snapshotPurge": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=snapshotPurge",
                "snapshotRevert": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=snapshotRevert",
                "updateDataLocality": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=updateDataLocality",
                "updateReplicaCount": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f?action=updateReplicaCount"
            },
            "backingImage": "",
            "backupStatus":
            [],
            "conditions":
            {
                "restore":
                {
                    "lastProbeTime": "",
                    "lastTransitionTime": "2021-11-23T08:59:57Z",
                    "message": "",
                    "reason": "",
                    "status": "False",
                    "type": "restore"
                },
                "scheduled":
                {
                    "lastProbeTime": "",
                    "lastTransitionTime": "2021-11-23T08:59:57Z",
                    "message": "",
                    "reason": "",
                    "status": "True",
                    "type": "scheduled"
                },
                "toomanysnapshots":
                {
                    "lastProbeTime": "",
                    "lastTransitionTime": "2021-11-23T08:59:57Z",
                    "message": "",
                    "reason": "",
                    "status": "False",
                    "type": "toomanysnapshots"
                }
            },
            "controllers":
            [
                {
                    "actualSize": "0",
                    "address": "",
                    "currentImage": "longhornio/longhorn-engine:v1.1.3-rc1",
                    "endpoint": "",
                    "engineImage": "longhornio/longhorn-engine:v1.1.3-rc1",
                    "hostId": "c1-worker2",
                    "instanceManagerName": "instance-manager-e-adfe6ed8",
                    "isExpanding": false,
                    "lastExpansionError": "",
                    "lastExpansionFailedAt": "",
                    "lastRestoredBackup": "",
                    "name": "pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f-e-d539e617",
                    "requestedBackupRestore": "",
                    "running": false,
                    "size": "12582912"
                }
            ],
            "created": "2021-11-23 08:59:51 +0000 UTC",
            "currentImage": "longhornio/longhorn-engine:v1.1.3-rc1",
            "dataLocality": "disabled",
            "disableFrontend": false,
            "diskSelector": null,
            "engineImage": "longhornio/longhorn-engine:v1.1.3-rc1",
            "fromBackup": "",
            "frontend": "blockdev",
            "id": "pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f",
            "kubernetesStatus":
            {
                "lastPVCRefAt": "",
                "lastPodRefAt": "",
                "namespace": "default",
                "pvName": "pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f",
                "pvStatus": "Bound",
                "pvcName": "simple-pvc-2",
                "workloadsStatus": null
            },
            "lastAttachedBy": "",
            "lastBackup": "",
            "lastBackupAt": "",
            "links":
            {
                "self": "http://192.168.64.79:30707/v1/volumes/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f"
            },
            "migratable": false,
            "name": "pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f",
            "nodeSelector": null,
            "numberOfReplicas": 1,
            "purgeStatus": null,
            "ready": true,
            "rebuildStatus":
            [],
            "recurringJobs": null,
            "replicas":
            [
                {
                    "address": "10.42.1.6",
                    "currentImage": "longhornio/longhorn-engine:v1.1.3-rc1",
                    "dataPath": "/var/lib/longhorn/replicas/pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f-d3c0b1ee",
                    "diskID": "38852b59-dacc-4aae-bc63-666dfe17fcf4",
                    "diskPath": "/var/lib/longhorn/",
                    "engineImage": "longhornio/longhorn-engine:v1.1.3-rc1",
                    "failedAt": "",
                    "hostId": "c1-worker1",
                    "instanceManagerName": "instance-manager-r-6bbd632e",
                    "mode": "",
                    "name": "pvc-180265a8-f69a-47c9-b89a-7fc48cbae35f-r-414dc9b3",
                    "running": true
                }
            ],
            "restoreRequired": false,
            "restoreStatus":
            [],
            "revisionCounterDisabled": false,
            "robustness": "unknown",
            "shareEndpoint": "",
            "shareState": "",
            "size": "12582912",
            "staleReplicaTimeout": 2880,
            "standby": false,
            "state": "attached",
            "type": "volume"
        }
```

</Details>

If I leave the volume long enough the node down icon will be shown next to the volume,
![image](https://user-images.githubusercontent.com/602540/143043336-18b57956-f012-434e-8599-05389742af08.png)

When click into it, will actually show as Unkown,
![image](https://user-images.githubusercontent.com/602540/143043762-6e2e025a-6624-40af-b809-6b5aac94998a.png)

**Comment by innobead** [comment]
This is not a bug because its replica actually still running, but its engine missing so needs users' intervention to fix it.

As @khushboo-rancher said, there is an icon showing the volume having some problem due to its engine/frontend, so the message is also correct and healthy status shows `unknown` correctly.

![image](https://user-images.githubusercontent.com/899492/143163217-13665ddc-69ab-43e4-842f-bc4f1b06acbd.png)
 
@kaxing Any other concerns actually?

**Comment by kaxing** [comment]
@innobead Just a slightly thing to be improved here: the "state" in volume list and the volume page are not consistent.
Checking with frontend dev at the moment.

**Comment by innobead** [comment]
@smallteeths This should be fixed in the master and 1.1.x branches as well. Thanks.
